From 812e979fe66e9c1a768fbcee3f568234fec2cded Mon Sep 17 00:00:00 2001
From: Franck Bui <fbui@suse.com>
Date: Tue, 15 Nov 2022 09:04:42 +0100
Subject: [PATCH] meson: install test-kernel-install only when
 -Dkernel-install=true

This patch fixes the following build failure:

  meson.build:3853:8: ERROR: Unknown variable "test_kernel_install_sh".

Fixes #25432.

(cherry picked from commit cc77a56532ddb59770e8312fc1b9954b0b135e72)

Related: #2138081
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 015849af49..35704947e3 100644
--- a/meson.build
+++ b/meson.build
@@ -3841,7 +3841,7 @@ exe = custom_target(
         install_dir : bindir)
 public_programs += exe
 
-if want_tests != 'false'
+if want_tests != 'false' and want_kernel_install
         test('test-kernel-install',
              test_kernel_install_sh,
              args : [exe.full_path(), loaderentry_install])
